File: sample-kanban-jvm/src/main/kotlin/sample/testcontainer/kanban/models/Task.kt
package sample.testcontainer.kanban.models

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "task", schema = "kanban")
data class Task(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,
    var description: String? = null,
    @Column(updatable = false)
    @Temporal(TemporalType.TIMESTAMP)
    var created: LocalDateTime? = null,
    @ManyToOne
    var status: Status? = null,
    @OneToMany
    @JoinColumn(name = "task_id")
    @OrderBy("created desc")
    var messages: List<Message>? = null,
    @ManyToMany
    @JoinTable(
        schema = "kanban",
        name = "task_person",
        joinColumns = [JoinColumn(name = "task_id")],
        inverseJoinColumns = [JoinColumn(name = "person_id")],
    )
    @OrderBy("created desc")
    var people: MutableList<Person>? = null,
) {
    override fun toString(): String {
        return "Task(id=$id, description=$description, created=$created, status=$status)"
    }
}

File: sample-kanban-go/app/models/task.go
package models

import "time"

type Task struct {
	Id          int64      `db:"id" goqu:"skipinsert"`
	StatusId    int64      `db:"status_id"`
	Description string     `db:"description"`
	Created     *time.Time `db:"created" goqu:"omitnil,skipupdate"`
	Status      *Status
	Messages    *[]Message
	People      *[]Person
}

func (t *Task) SafeMessageCount() int {
	if t.Messages != nil {
		return len(*t.Messages)
	}
	return 0
}

func (t *Task) SafeMessageList() *[]Message {
	if t.Messages == nil {
		return &[]Message{}
	}
	return t.Messages
}

func (t *Task) SafePeopleCount() int {
	if t.People != nil {
		return len(*t.People)
	}
	return 0
}

func (t *Task) SafePeopleList() *[]Person {
	if t.People == nil {
		return &[]Person{}
	}
	return t.People
}

func (t *Task) MemberById(id int64) *Person {
	for _, person := range *t.SafePeopleList() {
		if person.Id == id {
			return &person
		}
	}
	return nil
}
